raster-foundry/raster-foundry
# Add search to project list page and project select modal

## Overview

This PR primarily adds search capabilities to the project list page and the project select modal.

It also renames the `toolSearch` component to just `search`, as it is resource independent, and adds optional auto-focus to the search input.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo
Search input on projects page
![image](https://user-images.githubusercontent.com/2442245/31250267-51024a00-a9e8-11e7-974f-16270d06eaef.png)

Searching on projects page
![image](https://user-images.githubusercontent.com/2442245/31250296-65404882-a9e8-11e7-93ee-0c7c842c1f5d.png)


Search input in project select modal
![image](https://user-images.githubusercontent.com/2442245/31250245-3d5408e0-a9e8-11e7-82ca-f12ef41242ff.png)


Searching within the project select modal
![image](https://user-images.githubusercontent.com/2442245/31250209-1f2ffd4c-a9e8-11e7-9da6-0e3b1fded80e.png)

## Testing Instructions

 * Go to the project list page and try searching (should be automatically focused).
 * Open a tool and when selecting a project for a source node, try searching (should also autofocus).

Closes #2348
